### PR TITLE
[O11y][Spring Boot] Update README to use documentation guidelines.

### LINKED ISSUE #9035: Apply the Integrations doc template to Spring Boot
The Spring Boot integration went GA in 8.12.
The [documentation page](https://docs.elastic.co/integrations/spring_boot) should follow the Integrations [doc template](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#overview).

### PR DESCRIPTION
**Pull request by harnish-elastic**
## Proposed commit message

- Updated README as per the [doc template](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#overview).

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes #9035

### CODE REVIEW

File: packages/spring_boot/_dev/build/docs/README.md [hunk 1]
```diff
@@ -1,17 +1,47 @@
 # Spring Boot integration
 
+## Overview
+
 The Spring Boot integration is used to fetch observability data from [Spring Boot Actuator web endpoints](https://docs.spring.io/spring-boot/docs/2.6.3/actuator-api/htmlsingle/) and ingest it into Elasticsearch.
 
+Use the Spring Boot integration to:
+
+- Collect logs related to the auditevents and httptrace and metrics related to the gc, memory and threading.
+- Create visualizations to monitor, measure and analyze the usage trend and key data, and derive business insights.
```
Comment (kush-elastic) [inline packages/spring_boot/_dev/build/docs/README.md:10]:
```suggestion
- Use dashboards to visualize, monitor, measure, and analyze the usage trend and key data, and derive business insights.
```

File: packages/spring_boot/_dev/build/docs/README.md [hunk 2]
```diff
@@ -1,17 +1,47 @@
 # Spring Boot integration
 
+## Overview
+
 The Spring Boot integration is used to fetch observability data from [Spring Boot Actuator web endpoints](https://docs.spring.io/spring-boot/docs/2.6.3/actuator-api/htmlsingle/) and ingest it into Elasticsearch.
 
+Use the Spring Boot integration to:
+
+- Collect logs related to the auditevents and httptrace and metrics related to the gc, memory and threading.
```
Comment (kush-elastic) [inline packages/spring_boot/_dev/build/docs/README.md:9]:
```suggestion
- Gather logs pertaining to auditevents, httptraces, and metrics associated with garbage collection, memory, and threading.
```

File: packages/spring_boot/_dev/build/docs/README.md [hunk 3]
```diff
@@ -1,17 +1,47 @@
...
+Logs help you keep a record of events that happen on your machine. The `Log` data streams collected by Spring Boot integration are `auditevents` and `httptrace`, so that users can keep track of authentication events, HTTP request and response details, enabling comprehensive monitoring and security auditing.
+
+Metrics give you insight into the statistics of the Spring Boot. Metrics data streams collected by the Spring Boot integration include auditevents, gc, httptrace, memory and threading so that the user can monitor and troubleshoot the performance of the Spring Boot instances.
+
+Data streams:
+- `auditevents`: Collects information related to the authentication status, remote address, document ID and principal.
+- `gc`: Collects information related to the gc collector name, memory usage before and after collection, thread count and time metrics.
+- `httptrace`: Collects information related to the http requests, status response, principal and session details.
+- `memory`: Collects information related to the heap and non-heap memory, buffer pool and manager.
+- `threading`: Collects information related to the thread allocations, monitoring and CPU times.
```
Comment (kush-elastic) [inline packages/spring_boot/_dev/build/docs/README.md:26]:
```suggestion
Data streams:
- `auditevents`: Aggregates data on authentication outcomes, client IP addresses, document identifiers, and user principals.
- `gc`: Gathers metrics on garbage collection events, including collector names, memory utilization pre- and post-collection, thread counts, and duration metrics.
- `httptrace`: Captures details of HTTP transactions, including request paths, response statuses, user principals, and session identifiers.
- `memory`: Accumulates data on memory usage, differentiating between heap and non-heap memory spaces, buffer pools, and memory management.
- `threading`: Collects insights into thread creation, runtime monitoring, and CPU utilization metrics.
```

**Review by kush-elastic — CHANGES_REQUESTED** [review]

**Review by kush-elastic — APPROVED** [review]
LGTM!

File: packages/spring_boot/_dev/build/docs/README.md [hunk 4]
```diff
@@ -1,17 +1,47 @@
...
+Note:
+- Users can monitor and see the log inside the ingested documents for Spring Boot in the `logs-*` index pattern from `Discover`, and for metrics, the index pattern is `metrics-*`.
+
 ## Compatibility
 
 This integration has been tested against Spring Boot v2.7.17 with LTS JDK versions 8, 11, 17, and 21.
 
+## Prerequisites
+
+You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it. You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended or self-manage the Elastic Stack on your own hardware.
```
Comment (alaudazzi) [inline packages/spring_boot/_dev/build/docs/README.md:37]: I would suggest to:
1. Move this content to the section "Requirements"
2. Remove the section Prerequisites.
3. Put a comma after the text `which is recommended`

File: packages/spring_boot/_dev/build/docs/README.md [hunk 5]
```diff
@@ -1,17 +1,47 @@
...
+
 ## Requirements
 
 In order to ingest data from Spring Boot:
 - You must know the host for Spring Boot application, add that host while configuring the integration package.
 - Add default path for jolokia.
-- Spring-boot-actuator module provides all Spring Boot’s production-ready features. So add below dependency in `pom.xml` file.
+- Spring-boot-actuator module provides all Spring Boot's production-ready features. So add below dependency in `pom.xml` file.
```
Comment (alaudazzi) [inline packages/spring_boot/_dev/build/docs/README.md:44]:
```suggestion
- Spring-boot-actuator module provides all Spring Boot's production-ready features. You also need to add the following dependency to the `pom.xml` file:
```

File: packages/spring_boot/_dev/build/docs/README.md [hunk 6]
```diff
@@ -28,11 +58,19 @@ In order to ingest data from Spring Boot:
...
 
+## Setup
+
+For step-by-step instructions on how to set up an integration, see the [Getting Started](https://www.elastic.co/guide/en/welcome-to-elastic/current/getting-started-observability.html) guide.
+
+## Validation
+
+After the integration is successfully configured, click on the *Assets* tab of the Spring Boot Integration to display the available dashboards. Select the dashboard for your configured data stream, which should be populated with the required data.
+
 ### Troubleshooting
```
Comment (alaudazzi) [inline packages/spring_boot/_dev/build/docs/README.md:69]:
```suggestion
## Troubleshooting
```
As per documentation guidelines, this should be a H2

**Review by alaudazzi — CHANGES_REQUESTED** [review]
LGTM after comments are integrated.

**Review by alaudazzi — APPROVED** [review]
LGTM

File: packages/spring_boot/docs/README.md [hunk 7]
```diff
@@ -1,17 +1,45 @@
...
+
+Data streams:
+- `auditevents`: Collects information related to the authentication status, remote address, document ID and principal.
+- `gc`: Collects information related to the gc collector name, memory usage before and after collection, thread count and time metrics.
+- `httptrace`: Collects information related to the http requests, status response, principal and session details.
+- `memory`: Collects information related to the heap and non-heap memory, buffer pool and manager.
+- `threading`: Collects information related to the thread allocations, monitoring and CPU times.
+
+Note:
+- Users can monitor and see the log inside the ingested documents for Spring Boot in the `logs-*` index pattern from `Discover`, and for metrics, the index pattern is `metrics-*`.
```
[the remark this answers is not in the document]
Comment (harnish-elastic) [reply packages/spring_boot/docs/README.md]: Strongly feel to add quotes to highlight keywords like `logs-*`, `Discover`, and `metrics-*`. Other than that accepted suggestions.

**Review by devamanv — APPROVED** [review]
LGTM

**Pull request merged**